pandas/core/groupby/ops.py

```diff
@@ -165,26 +165,37 @@ def apply(self, f, data, axis=0):
         mutated = self.mutated
         splitter = self._get_splitter(data, axis=axis)
         group_keys = self._get_group_keys()
-
+        reuse_result = False
```

Comment:
this use of a flag is very odd and not very pythonic why is this needed at all?

Reply:
if you really really need this, then simply set ``result_values=None`` and test this

Reply:
I've rewritten the logic to neither require the `status` identifier nor the `reuse_result` flag. The `reuse_result` flag was necessary in the previous implementation to catch the case for `MultiIndex` data which is handled by raising an exception in Cython. This could be handled differently if the multiindex check is performed outside of Cython but I didn't want to change too much about the current behavior. Having the index check close to the implementation seemed like a good idea to me